### PR TITLE
Throw custom error on invalid package name

### LINKED ISSUE #21: Invalid name reported as available
```js
require('npm-name')('_AVA').then(console.log) //true
```
But such names [are invalid](https://docs.npmjs.com/files/package.json#name). So it must be false.

### PR DESCRIPTION
**Pull request by astur**
Now it throws error when package name is invalid

Fixes #21

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,8 +3,17 @@ const got = require('got');
 const isScoped = require('is-scoped');
 const registryUrl = require('registry-url')();
 const zip = require('lodash.zip');
+const validate = require('validate-npm-package-name');
 
 function request(name) {
+	const isValid = validate(name);
+	if (!isValid.validForNewPackages) {
+		const err = new Error(`Invalid package name: ${name}`);
```
Comment (sindresorhus) [inline index.js]: Better to use a custom error class so we can more easily and reliably detect it in `npm-name-cli`.

```js
class InvalidNameError extends Error {}
```

**Comment by sindresorhus** [comment]
Thanks! Can you give the PR a proper title?

File: index.js (hunk 1 above)
Comment (astur) [reply index.js]: Done in https://github.com/sindresorhus/npm-name/pull/22/commits/7bf87d62de21b195ea941df9ad4d083c7b40fa8b

File: index.js [hunk 2]
```diff
@@ -3,8 +3,17 @@ const got = require('got');
...
+		err.warnings = isValid.warnings;
+		err.errors = isValid.errors;
```
[the remark this answers is not in the document]
Comment (astur) [reply index.js]: Done in https://github.com/sindresorhus/npm-name/pull/22/commits/949097f78166be1d3ec18210048afa2f3b3ee13d

**Comment by sindresorhus** [comment]
Looks good. Thanks :)

**Pull request merged**